pkg/manager/impl/execution_manager_test.go

```diff
@@ -2464,6 +2464,111 @@ func TestAssignResourcesIfUnset(t *testing.T) {
 	}, assignedResources)
 }
 
+func TestCheckTaskRequestsLessThanLimits(t *testing.T) {
+	ctx := context.Background()
+	identifier := &core.Identifier{
+		ResourceType: core.ResourceType_TASK,
+		Project:      project,
+		Domain:       domain,
+		Name:         name,
+		Version:      version,
+	}
+	t.Run("use_limit", func(t *testing.T) {
+		resources := &core.Resources{
+			Requests: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "1",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "2",
+				},
+			},
+			Limits: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "2",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1",
+				},
+			},
+		}
+		checkTaskRequestsLessThanLimits(ctx, identifier, resources)
+		assert.True(t, proto.Equal(&core.Resources{
+			Requests: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "1",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1",
+				},
+			},
+			Limits: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "2",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1",
+				},
+			},
+		}, resources))
+	})
+	t.Run("nothing_to_override", func(t *testing.T) {
+		resources := &core.Resources{
+			Requests: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "2",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1",
+				},
+			},
+			Limits: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "2",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1.5",
+				},
+			},
+		}
+		checkTaskRequestsLessThanLimits(ctx, identifier, resources)
+		assert.True(t, proto.Equal(&core.Resources{
+			Requests: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "2",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1",
+				},
+			},
+			Limits: []*core.Resources_ResourceEntry{
+				{
+					Name:  core.Resources_CPU,
+					Value: "2",
+				},
+				{
+					Name:  core.Resources_MEMORY,
+					Value: "1.5",
+				},
+			},
+		}, resources))
+	})
```

Comment:
Can you add a test that tests this logic? Specifically, can you add a test where one of the resource does not have the limit specified? 
https://github.com/lyft/flyteadmin/pull/126/files#diff-fc047e54b9dd82ca7c89ac9b32cb07b3R282-R288

Reply:
at the stage where the method is called all values will have already been substituted with the application defaults. this is why my test only verifies that if the defaults are weirdly configured we choose the min of acceptable values

Reply:
oh interesting. So the control flow should never enter this block?
https://github.com/lyft/flyteadmin/pull/126/files#diff-fc047e54b9dd82ca7c89ac9b32cb07b3R283-R288

Reply:
unless the user is specifying a resource we don't provide defaults for. but in that case we don't find ourselves in the original conundrum where the substituted request > limit since there will be no limit default value which is a permissible situation for scheduling

Reply:
From a unit-test point of view I think that it would still be nice to have a test for it since we don't know if in the future this function would be invoked through other code paths where the default limits were not pre-filled. But as you said it is not critical here.